src/rgw/rgw_rados.cc

```diff
@@ -9157,6 +9245,13 @@ int RGWRados::check_disk_state(librados::IoCtx io_ctx,
       // is already deleted as the `astate` testified.
       cls_rgw_encode_suggestion(CEPH_RGW_REMOVE | suggest_flag, list_state,
                                 suggested_updates);
+      if (suggest_flag) {
+        bilog_handler.add_maybe_flush(CLS_RGW_OP_ADD,
+                                      list_state,
+                                      get_zones_trace(svc.zone->get_zone(),
+                                                      bucket_info.bucket,
+                                                      nullptr));
```

Comment:
The `check_disk_state()` paths miss `zones_trace` usually saw when modifying BILog. Passing `nullptr` here, so the returned `rgw_zone_set` instance will have single item only: the one constructed basing on `RGWRados::svc.zone->get_zone()`. Is this enough?